Fotoblog.BLL/Services/PhotosService/PhotoService.cs

```diff
@@ -0,0 +1,185 @@
+﻿using AutoMapper;
+using Fotoblog.BLL.Services.ServiceResultNS;
+using Fotoblog.DAL;
+using Fotoblog.DAL.Entities;
+using Fotoblog.Utils.ViewModels.Photos;
+using Microsoft.AspNetCore.Http;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Formats;
+using SixLabors.ImageSharp.Formats.Jpeg;
+using SixLabors.ImageSharp.Formats.Png;
+using SixLabors.ImageSharp.Processing;
+using System.IO;
+using System.Net;
+
+namespace Fotoblog.BLL.Services.PhotosService
+{
+    public class PhotoService : BaseService, IPhotoService
+    {
+        private readonly string[] _allowedExtensions = { ".jpg", "jpeg", ".png" };
+        private readonly string[] _allowedImgContentTypes = { "image/jpeg", "image/png" };
+        private string? _uploadedFileExtension;
+        private string _saveLocation = "c:\\uploads"; // random folder added later
+        private readonly int _thumbnailWidth = 600;
+        public PhotoService(ApplicationDbContext dbContext, IMapper mapper) : base(dbContext, mapper)
+        {
+        }
+
+        public async Task<ServiceResult> AddPhoto(NewPhotoDataVm newPhotoDataVm)
+        {
+            try
+            {
+                doInputValidation(newPhotoDataVm);
+                PreparePhotoSaveLocation();
+            }
+            catch(Exception)
+            {
+                return ServiceResult.Fail(ErrorCodes.CouldNotSaveTheFile);
+            }
+                
+
+            var origSavePath = Path.Combine(_saveLocation, $"orginal{_uploadedFileExtension}");
+            var thumbnailSavePath = Path.Combine(_saveLocation, $"thumbnail{_uploadedFileExtension}");
+
+            try
+            {
+                await saveOriginalPhoto(origSavePath, newPhotoDataVm.file);
+                await saveThumbnail(thumbnailSavePath, newPhotoDataVm.file);
```

Comment:
Do czego w tej aplikacji jest wykorzystywana miniaturka? Skoro tak czy inaczej trzymamy oryginalne zdjęcie to nie lepiej je skalować dopiero gdy zajdzie taka potrzeba?

Reply:
miniatury zdjęć będą wyświetlane w galerii. rozmiar oryginalny będzie doczytywany po kliknięciu miniatury. Myślę, że patrząc przez pryzmat wydajności lepiej jest przeskalować zdjęcie raz przy dodaniu i trzymać je na serwerze, niż skalować zdjęcia dla każdego requesta

Reply:
Pobieranie zdjęcia w wysokiej rozdzielczości przy każdym kliknięciu w miniaturę nie jest wydajne. Lepiej pobrać kolekcję zdjęć raz i je przeskalować na mniejsze (trzymając oryginalne w pamięci). W ten sposób bardziej obciążymy komputer klienta ale serwer nie będzie musiał odpowiadać na tak wiele zapytań.

Reply:
pogadamy o tym jak bedziesz miał czas